dynamicio/base.py

```diff
@@ -2,21 +2,20 @@
 
 """BaseResource class for creating various resources types."""
 from abc import ABC, abstractmethod
-from typing import Optional, Type, TypeVar
+from copy import deepcopy
+from typing import Optional, Type
 
 import pandas as pd
 import pandera as pa
 from pandera import SchemaModel
 from pydantic import BaseModel
 
-SchemaType = TypeVar("SchemaType", bound=pa.SchemaModel)  # TODO: utilise this
-
 
 class BaseResource(BaseModel, ABC):
```

Comment:
Actually, now that I think about it, it seems that `BaseResource` is both a configuration object, and it contains IO logic... Is this necessary? Could we not pass in a `pydantic` config object into the resource class instead of inheriting from `BaseModel`?

Reply:
So, this really is the crux of the matter. There are several ways to go about this. My first implementation was along the lines of create a resource (just a pydantic model, no functions) and then pass it to a read/write function that takes a  resource of a specific type. The reason I combined these into one, was to keep the use of it simple. We can maybe achieve this with different means.

@deaglancrew and I had a long discussion about using a creational builder pattern here, separationg out configuration, io, injection, validation and logging into their own classes. We did not yet find a technical solution for having a nice API for it and connecting the dots between injection and reading wasn't figured out either. Imo, there might be a tradeoff between usability+readability on one side and testability/solid on the other. Let's have a look at what we can do here in our pairing later.

Reply:
Maybe we could find some inspiration on github